### PR TITLE
Add user-supplied config file to gitignore

### PR DESCRIPTION
**Pull request by karima662**
excluding the first 3 commits, addressing #27

**logic of changes made:**
- save name of config file supplied to `get_config()` in `slushy_init()` as an attribute "config_file_name" to `config_list`
- call `config_file_name` from the `config` param in `slushy_init()`
- supply this to the newly added `update_ignores(additional_unignore_files = ...)` param
- within this function, an exclamation mark is prepended to `additional_unignore_files` and then appended to `renv_files`
- add tests for `get_config()` and `update_ignores()`

### CODE REVIEW

File: R/config.R [hunk 1]
```diff
@@ -14,8 +14,13 @@ get_config <- function(config_file = "slushy_config.yml"){
   # get name of default config
   config <- config::get(value = "config_name", config = "default", file = config_file)
```
Comment (bzkrouse) [inline R/config.R:15]:
```suggestion
  config <- get(value = "config_name", config = "default", file = config_file)
```

File: R/config.R [hunk 2]
```diff
@@ -14,8 +14,13 @@ get_config <- function(config_file = "slushy_config.yml"){
...
 
-  config::get(value = config, config = "default", file = config_file)
-
+  config_list <- config::get(value = config, config = "default", file = config_file)
```
Comment (bzkrouse) [inline R/config.R:17]:
```suggestion
  config_list <- get(value = config, config = "default", file = config_file)
```

File: R/config.R [hunk 3]
```diff
@@ -14,8 +14,13 @@ get_config <- function(config_file = "slushy_config.yml"){
...
+  
+  # attach the file name as an attribute
+  attr(config_list, "config_file_name") <- config_file
```
Comment (bzkrouse) [inline R/config.R:20]:
```suggestion
  structure(config_list,
                 config_file_name = config_file)
```

File: R/config.R [hunk 4]
```diff
@@ -14,8 +14,13 @@ get_config <- function(config_file = "slushy_config.yml"){
...
+  
+  # return the configuration list with the attached attribute
+  return(config_list)
```
Comment (bzkrouse) [inline R/config.R:23]:
```suggestion
```

File: R/slushy_trim.R [hunk 5]
```diff
@@ -31,7 +31,7 @@
 #'  slushy_trim(config = get_config("slushy_config.yml"))
 #' }
 #' 
-slushy_trim <- function(project = NULL, config = get_config(config_file = "slushy_config.yml")) {
+slushy_trim <- function(project = NULL, config = get_config()) {
```
Comment (bzkrouse) [inline R/slushy_trim.R:34]: I like this, can we add it to the other functions as well?

**Review by bzkrouse — CHANGES_REQUESTED** [review]
looks great, just a couple changes :)

File: R/slushy_trim.R (hunk 5 above)
Comment (karima662) [reply R/slushy_trim.R:34]: just to clarify - would you prefer to have `slushy_config.yml` as the default or empty (for each function)?

Comment (bzkrouse) [reply R/slushy_trim.R:34]: empty is good, since get_config is an exported function so people can easily see the default!

**Review by bzkrouse — CHANGES_REQUESTED** [review]
Looks great!!! Can you please also update the NEWS file since this is a user-facing change?

File: NEWS.md [hunk 6]
```diff
@@ -1,3 +1,8 @@
+# slushy development 0.5.2.9001
+- Add user-supplied configuration file to be in the `gitignore` file
```
Comment (bzkrouse) [inline NEWS.md:2]: just this line to describe the user-facing change is perfect, thank you :) You can remove the other lines.

File: NEWS.md [hunk 7]
```diff
@@ -1,3 +1,6 @@
+# slushy development 0.5.2.9001
+- Add user-supplied configuration file to be 'unignored' in the `gitignore` file
+
 # slushy development 0.5.2.9000
```
Comment (bzkrouse) [inline NEWS.md:4]:
```suggestion
```

File: R/config.R [hunk 8]
```diff
@@ -12,10 +12,15 @@ get_config <- function(config_file = "slushy_config.yml"){
...
-  config::get(value = config, config = "default", file = config_file)
+  config <- get(value = "config_name", config = "default", file = config_file)
 
+  config_list <- get(value = config, config = "default", file = config_file)
+  
+  # attach the file name as an attribute
+  structure(config_list,
+                 config_file_name = config_file)
+  
+  # return the configuration list with the attached attribute
```
Comment (bzkrouse) [inline R/config.R:23]:
```suggestion
  # return the configuration list with the attached attribute
```

**Review by bzkrouse — APPROVED** [review]

**Pull request merged**